code/modules/mob/living/simple_animal/abnormality/waw/snow_whites_apple.dm

```diff
@@ -150,21 +126,59 @@ GLOBAL_LIST_EMPTY(vine_list)
 		togglemovement = TRUE
 		switch(chosen_attack)
 			if(1)
-				if(vine_cooldown <= world.time)
+				if(unique_cooldown <= world.time)
 					vinespike()
 		return
 
-	if(vine_cooldown <= world.time)
+	if(unique_cooldown <= world.time)
 		vinespike()
 
+/mob/living/simple_animal/hostile/abnormality/snow_whites_apple/proc/TryTeleport() //stolen from knight of despair
+	dir = 2
+	if(teleport_cooldown > world.time)
+		return FALSE
+	teleport_cooldown = world.time + teleport_cooldown_delay
+	var/list/teleport_potential = list()
+	for(var/turf/T in GLOB.xeno_spawn)
+		teleport_potential += T
+	if(!LAZYLEN(teleport_potential))
+		return FALSE
+	var/turf/teleport_target = pick(teleport_potential)
+	animate(src, alpha = 0, time = 5)
+	new /obj/effect/temp_visual/guardian/phase(get_turf(src))
+	SLEEP_CHECK_DEATH(5) // TODO: Add some cool effects here
+	animate(src, alpha = 255, time = 5)
+	new /obj/effect/temp_visual/guardian/phase/out(teleport_target)
+	forceMove(teleport_target)
+	SpreadPlants()
+	oldGrowth()
+
+/mob/living/simple_animal/hostile/abnormality/snow_whites_apple/proc/SpreadPlants()
+	if(!isturf(loc) || isspaceturf(loc))
+		return
+	if(locate(/obj/structure/apple_vine) in get_turf(src))
+		return
+	new /obj/structure/apple_vine(loc)
+
 /mob/living/simple_animal/hostile/abnormality/snow_whites_apple/proc/vinespike()
-	for(var/obj/structure/alien/weeds/apple_vine/W in view(18, src))
-		W.vinespike()
-	vine_cooldown = world.time + vine_cooldown_time
+	for(var/mob/living/L in livinginview(vision_range, src))
+		for(var/obj/structure/apple_vine/W in L.loc)
```

Comment:
You can use locate() instead. Additionally, use get_turf(L) instead of L.loc

Reply:
Will do thank you. I was unsure if it helped with processing power.